server/service/getcars.ts

```diff
@@ -0,0 +1,81 @@
+import {db} from './db';
+import {Express} from "express";
+import {ICars} from "./interfaces";
+
+export const GetAllAsync = (app: Express) => {
+    const cars: ICars ={}
+    const drives = "SELECT drivetype FROM transmissions"
+    const made = "SELECT * FROM carbrands"
+    const gbs = "SELECT gb FROM gearboxes"
+    const eng = "SELECT enginetype FROM enginetypes"
+    const bdt = "SELECT bodytype FROM bodytypes"
+    app.get('/search', async (req,res)=>{
+        await Promise.all([
+            new Promise<ICars['brands']>((resolve, reject)=> {
+                db.query(made, (err, data: ICars['brands']) => {err? reject(err):resolve(data)})
+            }),
+            new Promise<ICars['trans']>((resolve, reject)=> {
+                db.query(drives, (err, data: ICars['trans']) => {err? reject(err):resolve(data)})
+            }),
+            new Promise<ICars['gears']>((resolve, reject)=> {
+                db.query(gbs, (err, data: ICars['gears']) => {err? reject(err):resolve(data)})
+            }),
+            new Promise<ICars['engines']>((resolve, reject)=> {
+                db.query(eng, (err, data: ICars['engines']) => {err? reject(err):resolve(data)})
+            }),
+            new Promise<ICars['bodies']>((resolve, reject)=> {
+                db.query(bdt, (err, data: ICars['bodies']) => {err? reject(err):resolve(data)})
+            })
+        ]).then((values)=>{
+            cars.brands = values[0]
```

Comment:
dont forget semicolons. If u have errors it may save time u to find place with error